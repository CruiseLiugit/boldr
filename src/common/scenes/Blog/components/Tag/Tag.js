/* @flow */
import React from 'react';
import Link from 'react-router/lib/Link';
import Avatar from 'material-ui/Avatar';
import TagIcon from 'material-ui/svg-icons/maps/local-offer';
import Chip from 'material-ui/Chip';

const Tag = (props: { name: string }) => {
  return (
  <Link to={ `/blog/tags/${props.name}` }>
    <Chip>
        <Avatar icon={ <TagIcon /> } />
          { props.name }
      </Chip>
  </Link>);
};

export default Tag;
