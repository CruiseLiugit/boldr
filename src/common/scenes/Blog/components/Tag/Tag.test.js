import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Tag from './Tag';

describe('<Tag />', () => {
  it('accepts props and renders them.', () => {
    const wrapper = shallow(
      <Tag name="imATag" />);
    expect(wrapper.instance().props.name).toBe('imATag');
  });
  it('renders snapshot', () => {
    const wrapper = shallow(<Tag />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
