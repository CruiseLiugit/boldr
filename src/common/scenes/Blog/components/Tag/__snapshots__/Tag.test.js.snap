exports[`<Tag /> renders snapshot 1`] = `
<Link
  onlyActiveOnIndex={false}
  style={Object {}}
  to="/blog/tags/undefined">
  <Label
    image={true}>
    <Icon
      as="i"
      name="tag" />
  </Label>
</Link>
`;
